core/src/consensus/fork_choice.rs

```diff
@@ -54,6 +64,35 @@ pub trait ForkChoice {
     ) -> Vec<Self::ForkChoiceKey>;
 }
 
+fn last_vote_able_to_land(
+    reset_bank: Option<&Bank>,
+    progress: &ProgressMap,
+    tower: &Tower,
+) -> bool {
+    let Some(heaviest_bank_on_same_voted_fork) = reset_bank else {
```

Comment:
It would be nice to add comments to each leg. I don't fully understand how to describe the reasoning concisely tho